data-tck/src/main/java/io/micronaut/data/tck/repositories/BookRepository.java

```diff
@@ -172,4 +173,6 @@ protected Book newBook(Author author, String title, int pages) {
     abstract List<Book> findByTitleInAndTotalPagesGreaterThan(List<String> titles, int totalPages);
 
     abstract Long countByTitleInAndTotalPagesGreaterThan(List<String> titles, int totalPages);
+
+    abstract Optional<BookDtoWithAuthorDto> queryByTitleContains(String title);
```

Comment:
Having 'AuthorDto` in 'BookDto` should work but compiler gives this error
> error: Unable to implement Repository method: PostgresBookRepository.queryByTitleContains(String title). Property [author] of type [io.micronaut.data.tck.entities.AuthorDTO] is not compatible with equivalent property of type [io.micronaut.data.tck.entities.Author] declared in entity: io.micronaut.data.tck.entities.Book

So we might need to check nested types if compatible. I will commit potential fix for this, but still `SELECT` does not use this joined entity and mapper also does not consider it when reading data.